src/users/usersActions.js

```diff
@@ -25,17 +26,42 @@ export const reportPresence = (isActive: boolean = true, newUserInput: boolean =
   });
 };
 
-const typingWorker = auth => ({
-  get_current_time: () => new Date().getTime(),
+const typingWorker = (state: GlobalState) => {
+  const auth: Auth = getAuth(state);
 
-  notify_server_start: (user_ids_array: number[]) => {
-    api.typing(auth, JSON.stringify(user_ids_array), 'start');
-  },
+  // User ID arrays are only supported in server versions >= 2.0.0-rc1, for
+  // versions before this, email arrays are used. If current server version is
+  // undetermined, user ID arrays are optimistically used.
+  // Support for user ID arrays was added in zulip/zulip commit 2f634f8.
```

Comment:
nit: use 9 hex digits for a commit ID (`git config --global core.abbrev 9` helps make this easy.)

For discussion see https://github.com/zulip/zulip-mobile/blob/master/docs/style.md#mentioning-commits , which I just wrote :stuck_out_tongue: to explain why I've done this, and recommended it, for a long time.